src/modules/sub-items/components/table-view/table.view.component.js

```diff
@@ -124,36 +245,55 @@ export default class TableViewComponent extends Component {
             headClass = `${headClass} ${headSortOrderClass} ${headSortByClass}`;
         }
 
-        const headerNameLabel = Translator.trans(/*@Desc("Name")*/ 'items_table.header.name', {}, 'sub_items');
-        const headerModifiedLabel = Translator.trans(/*@Desc("Modified")*/ 'items_table.header.modified', {}, 'sub_items');
-        const headerContentTypeLabel = Translator.trans(/*@Desc("Content type")*/ 'items_table.header.content_type', {}, 'sub_items');
-        const headerPriorityLabel = Translator.trans(/*@Desc("Priority")*/ 'items_table.header.priority', {}, 'sub_items');
-        const headerTranslationsLabel = Translator.trans(/*@Desc("Translations")*/ 'items_table.header.translations', {}, 'sub_items');
         const { selectedLocationsIds } = this.props;
+        const { columnsVisibility } = this.state;
         const anyLocationSelected = !!selectedLocationsIds.size;
 
         return (
             <thead className={headClass}>
                 <tr className="c-table-view__row">
-                    <td className={cellHeadClass}>
+                    <th className={`${cellHeadClass} ${cellClass}--checkbox`}>
                         <input type="checkbox" checked={anyLocationSelected} onChange={this.selectAll} />
-                    </td>
-                    <td className={cellHeadClass} />
-                    <td className={`${cellHeadClass} ${cellClass}--name ${cellSortClass}`} onClick={this.sortByName} tabIndex="-1">
-                        <span className="c-table-view__label">{headerNameLabel}</span>
-                    </td>
-                    <td className={`${cellHeadClass} ${cellClass}--date ${cellSortClass}`} onClick={this.sortByDate} tabIndex="-1">
-                        <span className="c-table-view__label">{headerModifiedLabel}</span>
-                    </td>
-                    <td className={cellHeadClass}>
-                        <span className="c-table-view__label">{headerContentTypeLabel}</span>
-                    </td>
-                    <td className={`${cellHeadClass} ${cellClass}--priority ${cellSortClass}`} onClick={this.sortByPriority} tabIndex="-1">
-                        <span className="c-table-view__label">{headerPriorityLabel}</span>
-                    </td>
-                    <td className={cellHeadClass} colSpan={2}>
-                        <span className="c-table-view__label">{headerTranslationsLabel}</span>
-                    </td>
+                    </th>
+                    <th className={cellHeadClass} />
+                    <th className={`${cellHeadClass} ${cellClass}--name ${cellSortClass}`} onClick={this.sortByName} tabIndex="-1">
+                        <span className="c-table-view__label">{this.headerLabels.name}</span>
+                    </th>
+                    {columnsVisibility.modified && (
```

Comment:
Please, create sub-rendering methods for such cases. It will improve code readability.